### ISSUE TITLE
Is there a way to restrict when the Health Checks run?

### ISSUE THREAD

**Issue by udlose** [Xabaril/AspNetCore.Diagnostics.HealthChecks#459]
I'd like to know how I can restrict the health checks to only run during a certain set of hours.  I have dependencies that have defined set of hourly downtimes - even in PROD.  If I can't restrict this I will be alerted for false positives.

The only way I can think of is to use `AddCheck<T>` and return `HealthCheckResult.Healthy()` in my custom HealthCheckRule during my off-hours using the approach described here [Implement health checks in ASP.NET Core services](https://docs.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/monitor-app-health#implement-health-checks-in-aspnet-core-services).  

I was wondering if this is already built-in or if my solution is the only way.

TIA.

**Comment by CarlosLanderas** [comment]
This custom user rules has been recently added for webhooks notifications. You have a configurable Func<UIReport,bool> to configure if a failure payload should be notified. Maybe we can add some kind of user configurable function to check if a healthcheck collection should be triggered, so the user can skip collections in a window time.

**Comment by udlose** [comment]
@CarlosLanderas is there an example of that anywhere?

**Comment by CarlosLanderas** [comment]
We have that implemented for Webhooks ( and samples as well), not implemented yet for healthcheck collector. I was suggesting what we could do :)

**Comment by udlose** [comment]
I pulled down the AspNetCored.Diagnostics.HealthChecks repo and looked for anything with a `Func<UIReport, bool>` signature.  I didn't find anything.  Could you kindly point me to where this is done?

**Comment by CarlosLanderas** [comment]
```csharp
setup
.AddWebhookNotification("webhook1",
    uri: "status/200?code=ax3rt56s", payload: "{ message: \"Webhook report for [[LIVENESS]]: [[FAILURE]] - Description: [[DESCRIPTIONS]]\"}",
    restorePayload: "{ message: \"[[LIVENESS]] is back to life\"}",
    shouldNotifyFunc: report => DateTime.UtcNow.Hour >= 8 && DateTime.UtcNow.Hour <= 23
    customMessageFunc: (report) =>
    {
        var failing = report.Entries.Where(e => e.Value.Status == UIHealthStatus.Unhealthy);
        return $"{failing.Count()} healthchecks are failing";
    },
    customDescriptionFunc: report =>
    {
        var failing = report.Entries.Where(e => e.Value.Status == UIHealthStatus.Unhealthy);
        return $"HealthChecks with names {string.Join("/", failing.Select(f => f.Key))} are failing";
    });
```

https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/master/doc/webhooks.md#configuring-webhooks-by-code

**Comment by CarlosLanderas** [comment]
@udlose , Are you thinking in something like this?

```csharp
services
.AddHealthChecksUI(setupSettings: setup =>
{     
    setup.ShouldExecuteHealthChecksCollectorCallback(serviceProvider=> DateTime.UtcNow.Hour >= 20 && DateTime.UtcNow.Hour < 24);
});
```

Or... another dummy sample:

```csharp
services
 .AddHealthChecksUI(setupSettings: setup =>
{
 setup.ShouldExecuteHealthChecksCollectorCallback(sp => {
    var scheduler = sp.GetRequiredService<IExecutorSchedule>();
    return scheduler.ShouldExecute();
});
```

**Comment by Lonli-Lokli** [comment]
@CarlosLanderas Was this ShouldExecuteHealthChecksCollectorCallback implemented?
